spec/lib/wos_records_spec.rb

```diff
@@ -0,0 +1,216 @@
+require 'htmlentities'
+
+describe WosRecords do
+  let(:encoded_records) { File.read('spec/fixtures/wos_client/wos_encoded_records.html') }
+  let(:decoded_records) do
+    coder = HTMLEntities.new
+    coder.decode(encoded_records)
+  end
+  let(:wos_records_encoded) { described_class.new(encoded_records: encoded_records) }
+  let(:wos_records_decoded) { described_class.new(records: decoded_records) }
+
+  let(:recordsA) do
+    <<-XML_A
+        <records>
+          <REC><UID>WOS:A1</UID></REC>
+          <REC><UID>WOS:A2</UID></REC>
+        </records>
+    XML_A
+  end
+  let(:recordsB) do
+    <<-XML_B
+        <records>
+          <REC><UID>WOS:A2</UID></REC>
+          <REC><UID>WOS:B2</UID></REC>
+        </records>
+    XML_B
+  end
+  let(:wos_recordsA) { described_class.new(records: recordsA) }
+  let(:wos_recordsB) { described_class.new(records: recordsB) }
+
+  describe '#new' do
+    it 'works with encoded records' do
+      result = described_class.new(encoded_records: encoded_records)
+      expect(result).to be_an described_class
+    end
+    it 'works with decoded records' do
+      result = described_class.new(records: decoded_records)
+      expect(result).to be_an described_class
+    end
+  end
+
+  describe '#count' do
+    it 'returns Integer' do
+      count = wos_records_encoded.count
+      expect(count).to be_an Integer
+    end
+    it 'delegates to rec_nodes.count' do
+      count = wos_records_encoded.count
+      expect(count).to eq wos_records_encoded.rec_nodes.count
+    end
+  end
+
+  describe '#doc' do
+    it 'works with encoded records' do
+      result = wos_records_encoded.doc
+      expect(result).to be_an Nokogiri::XML::Document
+    end
+    it 'works with decoded records' do
+      result = wos_records_decoded.doc
+      expect(result).to be_an Nokogiri::XML::Document
+    end
+  end
+
+  describe '#each' do
+    it 'yields WoS record elements' do
+      wos_records_encoded.all? { |rec| expect(rec).to be_an Nokogiri::XML::Element }
+    end
+  end
+
+  describe 'Enumerable mixin' do
+    it 'responds to Enumerable.instance_methods' do
+      methods = Enumerable.instance_methods
+      methods.each { |method| expect(wos_records_encoded).to respond_to(method) }
```

Comment:
There is no need to enumerate methods individually, because that is how Ruby inheritance works.  Please replace this spec with `described_class.is_a?(Enumerable)` or consider removing it.

Reply:
Doesn't work, i.e.
```
  1) WosRecords Enumerable mixin responds to Enumerable.instance_methods
     Failure/Error: expect(described_class).to be_an Enumerable
       expected WosRecords to be a kind of Enumerable
     # ./spec/lib/wos_records_spec.rb:74:in `block (3 levels) in <top (required)>'
```
```
  1) WosRecords Enumerable mixin responds to Enumerable.instance_methods
     Failure/Error: expect(described_class).to be_kind_of Enumerable
       expected WosRecords to be a kind of Enumerable
     # ./spec/lib/wos_records_spec.rb:74:in `block (3 levels) in <top (required)>'
```
Leave it as is?

Reply:
Never mind, got it working on an instance object.

Reply:
Done.